web_security_dialog/static/src/js/web_security_dialog.js

```diff
@@ -0,0 +1,120 @@
+odoo.define("web_security_dialog.SecurityDialog", function(require) {
+    "use strict";
+
+    var core = require('web.core');
+    var FromWidget = require('web.form_widgets');
+    var Dialog = require('web.Dialog');
+    var Model = require('web.DataModel');
+    var framework = require('web.framework');
+
+    var QWeb = core.qweb;
+    var _t = core._t;
+
+    FromWidget.WidgetButton.include({
+        init : function(field_manager, node) {
+            this._super(field_manager, node);
+            if(this.node.attrs.options) {
+                this.options = JSON.parse(this.node.attrs.options)
+                    ? JSON.parse(this.node.attrs.options)
+                            : false;
+                if(this.options) {
```

Comment:
if you use the above, options always will be an object, and then you can lose most of those checks here

Reply:
WIP